lib/rspec/sidekiq/matchers/have_enqueued_sidekiq_job.rb

```diff
@@ -1,11 +1,11 @@
 module RSpec
   module Sidekiq
     module Matchers
-      def have_enqueued_job(*expected_arguments)
-        HaveEnqueuedJob.new expected_arguments
+      def have_enqueued_sidekiq_job(*expected_arguments)
+        HaveEnqueuedSidekiqJob.new expected_arguments
       end
```

Comment:
> like figuring out how to make rspec-rails not include the conflicting matchers, or replace them

We can't do that because like you say, there would be argument errors.

If you'd like a better solution, include this in a major release, and include a constraint specifiying `rspec-rails < 3.4` in a minor release